### PR TITLE
feat: add timezone to date formatter

### LINKED ISSUE #419: [feature] Add option to pass default timezone
Message format has an option to pass default currency to the constructor like
```
import MessageFormat from '@messageformat/core';

const mf = new MessageFormat('en', {currency: 'EUR'});
```

Alongside for date and time formatting it might be necessary to set `timeZone` (or pass to Intl.DateTimeFormat). By default it uses "current" locale. However, sometimes (especially using lib in node.js or other backend) it might be very helpful. And it will be more convenient, rather than specifying a new custom formatter.

Example:
```
import MessageFormat from '@messageformat/core';

const mf = new MessageFormat('en', {timeZone: 'UTC'});
```

### PR DESCRIPTION
**Pull request by SkeLLLa**
Closes #419.

### CODE REVIEW

**Comment by SkeLLLa** [comment]
@eemeli here's a first part with modified formatter library. I've tried to make it as less breaking as possible. If these changes are ok, I can proceed with adding it to core lib.

File: packages/date-skeleton/src/get-date-formatter.test.ts [hunk 1]
```diff
@@ -116,6 +116,32 @@ describe('Options', () => {
...
 
+  test('UTC timezone', () => {
+    const onError = jest.fn();
+    const fmt = getDateFormatter('en-US-u-hc-h23', 'jms', 'UTC', onError);
+    // get target UTC time
+    const offset = date.getTimezoneOffset();
+    const targetTimestamp = new Date(date.toUTCString()).getTime();
+    const targetDate = new Date(targetTimestamp + offset * 60 * 1000);
+    const targetTimeString = targetDate.toTimeString().split(' ').shift();
+    expect(fmt(date)).toEqual(targetTimeString);
```
[the remark this answers is not in the document]
Comment (SkeLLLa) [reply packages/date-skeleton/src/get-date-formatter.test.ts:127]: Yes. Makes sense.

File: packages/date-skeleton/src/options.ts [hunk 2]
```diff
@@ -168,11 +168,16 @@ function compileOptions(
...
   onError: (error: DateFormatError) => void = error => {
     throw error;
   }
 ) {
-  const options: Intl.DateTimeFormatOptions = {};
+  const options: Intl.DateTimeFormatOptions = timeZone
+    ? {
+        timeZone
+      }
+    : {};
```
[the remark this answers is not in the document]
Comment (SkeLLLa) [reply packages/date-skeleton/src/options.ts:180]: Thanks, will do. With `null` I just followed the code with `currency` in number format to make it the same. But you're correct, without `null` the code will be simpler.

**Comment by SkeLLLa** [comment]
In addition I've added a test for currency.

And regarding it I have a side-question non related to this PR. Following code works as expected:

```ts
const mf = new MessageFormat('en', { currency: 'EUR' });
const src = '{value, number, currency}';
expect(mf.compile(src)({ value: 100 })).toBe('€100.00');
```

But this throws an error:
```ts
const mf = new MessageFormat('en', { currency: 'EUR' });
const src = '{value, number, ::currency}';
expect(mf.compile(src)({ value: 100 })).toBe('€100.00');
```

Is that expected?

**Comment by SkeLLLa** [comment]
@eemeli can you check if everything is ok here when you have time?
Thanks.

**Review by eemeli — APPROVED** [review]
This looks really good! Sorry for taking ages to get back to you, this fell off my radar somewhat.

**Comment by eemeli** [comment]
> [...] this throws an error:
> 
> ```ts
> const mf = new MessageFormat('en', { currency: 'EUR' });
> const src = '{value, number, ::currency}';
> expect(mf.compile(src)({ value: 100 })).toBe('€100.00');
> ```
> 
> Is that expected?

Yes. For skeletons, the currency code is required: https://github.com/unicode-org/icu/blob/main/docs/userguide/format_parse/numbers/skeletons.md#unit

**Comment by SkeLLLa** [comment]
@eemeli, no problem.  Thanks for an explanation.

Is there any release schedule or any approx date when it will be published?

Thanks.

**Comment by eemeli** [comment]
No fixed schedule, but I'll see if I can get a release out next weekend, or the one after.

**Pull request merged**